pylsp_ruff/ruff_lint.py

```diff
@@ -245,21 +237,15 @@ def load_config(workspace: Workspace, document: Document) -> dict:
         workspace.root_path, document.path, ["pyproject.toml"]
     )
 
-    # Load config from pyproject file if it exists
+    # Check if pyproject is present, ignore user settings if toml exists
     if pyproject_file:
-        try:
-            log.debug(f"Found pyproject file: {str(pyproject_file[0])}")
-
-            with open(str(pyproject_file[0]), "rb") as pyproject_toml:
-                toml_dict = tomllib.load(pyproject_toml)
-
-            toml_config = toml_dict.get("tool", {}).get("ruff", {})
-
-            # Update settings with local project settings
-            for key, value in toml_config.items():
-                settings[key] = value
-
-        except (tomllib.TOMLDecodeError, OSError) as e:
-            log.warning(f"Failed loading {str(pyproject_file)}: {e}")
+        log.debug(
+            f"Found pyproject file: {str(pyproject_file[0])}, "
+            + "skipping pylsp config."
+        )
+
+        # Reset settings to None except executable
+        settings = {key: None for key in settings.keys()}
+        settings["executable"] = _settings.get("executable", "ruff")
```

Comment:
It might be a bit clearer to just create settings in this if block, and add an else, rather than resetting it.

Reply:
I hadn't thought of that, nice suggestion :+1: